pulp_docker/app/viewsets.py

```diff
@@ -327,3 +327,47 @@ def create(self, request):
             }
         )
         return OperationPostponedResponse(result, request)
+
+
+class ManifestCopyViewSet(drf_viewsets.ViewSet):
+    """
+    ViewSet for copying manifests recursively.
+    """
+
+    serializer_class = serializers.ManifestCopySerializer
+
+    @swagger_auto_schema(
+        operation_description="Trigger an asynchronous task to copy manifests",
+        responses={202: AsyncOperationResponseSerializer},
+        request_body=serializers.ManifestCopySerializer,
+    )
+    def create(self, request):
+        """
+        Queues a task that creates a new RepositoryVersion by adding content units.
+        """
+        serializer = serializers.ManifestCopySerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        source_latest = serializer.validated_data['source_repository_version']
+        destination = serializer.validated_data['destination_repository']
+        content_manifests_in_repo = source_latest.content.filter(
+            _type="docker.manifest"
+        )
+        manifests_in_repo = models.Manifest.objects.filter(
+            pk__in=content_manifests_in_repo,
+        )
+        digests = request.data.get("digests")
+        media_types = request.data.get("media_types")
+        if digests is None:
+            manifests_to_add = manifests_in_repo
+        else:
+            manifests_to_add = manifests_in_repo.filter(digest__in=digests)
+        if media_types is not None:
+            manifests_to_add = manifests_to_add.filter(media_type__in=media_types)
```

Comment:
this will not work properly in case there will be specified digest and mediatype, this will just replace what was in manifests_too_add

Reply:
nevermind, i misread

Reply:
can we instead do a single query like this Manifest.objects.filter(digest__in=, media_type__in=) 
wdyt?

Reply:
I'd like to, but if digests or media_types are an empty list, manifests_to_add will be empty list. Got any ideas?

Reply:
after IRC discussion, this is ok

Reply:
@ipanova I improved this with the most recent push. wdyt?

Reply:
i like it :star: